### ISSUE TITLE
[MD057] Always do a case-sensitive check on all platforms

### ISSUE THREAD

**Issue by mschoettle** [rvben/rumdl#873]
### rumdl version

0.2.73

### Installation method

Pre-built binary

### Bug description

On macOS a relative link to `CHANGELOG.md` but the actual file `changelog.md` does not report `MD057`. On other platforms (such as `ubuntu-latest` on GitHub Actions) it is reported.

### Steps to reproduce

1. Create a file `changelog.md`
2. Create a file `test.md` with this content

    ```markdown
    # Test

    See the [changelog](CHANGELOG.md).
    ```
3. Run `rumdl check test.md`
4. See no error on macOS
5. See error "test.md:5:18: [MD057] Relative link 'CHANGELOG.md' does not exist" on Linux

For example within an alpine container (`python:alpine`: `docker run --rm -it python:alpine sh`):

```shell
pip install rumdl
touch changelog.md
printf "# Test\n\nSee the [changelog](CHANGELOG.md).\n" > test.md
rumdl check test.md
test.md:3:21: [MD057] Relative link 'CHANGELOG.md' does not exist

Issues: Found 1 issues in 1 file (14ms)
```

### Expected behavior

Also report a non-existent relative link on macOS.

### Actual behavior

See above

### Configuration

```toml

```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks @mschoettle!

Fixed in [202da50c](https://github.com/rvben/rumdl/commit/202da50c8a63751b10225ef23550369ae62035e1), released in [v0.2.74](https://github.com/rvben/rumdl/releases/tag/v0.2.74). MD057 now requires the link target’s exact case on case-insensitive filesystems too.

Closing as completed.
